website/project/model.py

```diff
@@ -755,6 +755,7 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     nodes = fields.AbstractForeignField(list=True, backref='parent')
     forked_from = fields.ForeignField('node', backref='forked', index=True)
     registered_from = fields.ForeignField('node', backref='registrations', index=True)
+    ultimate_parent = fields.StringField(index=True)
```

Comment:
After an architectural consult with Steve, please change this to be a `ForeignField` named `root`, and the `root` property to be named `_root`, as it's the cached field that we want to expose.